crates/m3/src/builder/table.rs

```diff
@@ -706,6 +706,33 @@ impl<F: TowerField> Table<F> {
 	pub fn is_power_of_two_sized(&self) -> bool {
 		matches!(self.table_size_spec, TableSizeSpec::PowerOfTwo)
 	}
+
+	/// Returns the abstract cost of this thing.
+	pub fn cost(&self) -> usize {
+		let mut per_tower_level: [usize; 7] = [0; 7];
+		for partition in self.partitions.values() {
+			let values_per_row = partition.values_per_row;
+			for zc in &partition.zero_constraints {
+				let expr = &zc.expr;
+				if expr.degree() == 0 {
+					continue;
+				}
+				let degree = expr.degree() - 1;
+				let n_mult_ops = expr.n_mult_ops();
+				per_tower_level[zc.tower_level] += n_mult_ops * degree * values_per_row;
+			}
+		}
+
+		let mut total_cost = 0;
+		for (tower_level, cost) in per_tower_level.iter().enumerate() {
+			total_cost += cost * (1 << tower_level).min(8);
```

Comment:
Leave a comment saying something like:

```
// Scale the constraint cost by the number of bits in the constrained elements.
// Treat the minimum number of bits as 8 because the B8 field is the smallest that we can
// evaluate zerocheck constraints over, using the univariate skip strategy.
```

Reply:
Oops, this was my first approach and I meant to remove it altogether!

Reply:
Oh, nvm, if it's unused then just delete, I missed that.